### PR TITLE
FPLA 776 update role access for gatekeeping state

### PR DESCRIPTION
**Pull request by lewisbirks**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: ccd-definition/CaseEvent.json [hunk 1]
```diff
@@ -447,13 +447,27 @@
     "Name": "Add statement of service (c9)",
     "Description": "Add statement of service",
     "DisplayOrder": 11,
-    "PreConditionState(s)": "Submitted",
-    "PostConditionState": "Submitted",
+    "PreConditionState(s)": "Gatekeeping",
+    "PostConditionState": "Gatekeeping",
```
Comment (akomisarek) [inline ccd-definition/CaseEvent.json]: Hmm a bit surprised why this change is here. Why do we need to change the state of add statement of service to gatekeeping?

File: e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    await I.completeEvent('Save and continue');
+    I.seeEventSubmissionConfirmation(config.administrationActions.sendToGatekeeper);
+    I.signOut();
+
+    await I.signIn(config.swanseaLocalAuthorityEmailUserOne, config.localAuthorityPassword);
+  }
+  await I.navigateToCaseDetails(caseId);
+});
+
+Scenario('local authority uploads documents', async (I, caseViewPage, uploadDocumentsEventPage) => {
```
Comment (akomisarek) [inline e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js]: As I can imagine we are already doing similar tests, maybe we could just reuse the steps instead of duplicating them? What do you think?

File: e2e/tests/localAuthorityMaintainsCaseAfterSubmission_test.js [hunk 3]
```diff
@@ -51,41 +50,3 @@ Scenario('local authority uploads court bundle', async (I, uploadDocumentsEventP
   caseViewPage.selectTab(caseViewPage.tabs.documents);
   I.seeDocument('Court bundle', 'mockFile.txt');
 });
-
-Scenario('local authority provides a statements of service', async (I, caseViewPage, loginPage, addStatementOfServiceEventPage) => {
```
Comment (akomisarek) [inline e2e/tests/localAuthorityMaintainsCaseAfterSubmission_test.js]: We are no longer doing statement of service in submission state?

File: ccd-definition/CaseEvent.json (hunk 1 above)
Comment (lewisbirks) [reply ccd-definition/CaseEvent.json]: We had a chat with Gail and she decided that it should be moved and was only in submitted before because the gatekeeping state didn't exist. We can double check with Gail before actually doing the PR.

File: e2e/tests/localAuthorityMaintainsCaseAfterSubmission_test.js (hunk 3 above)
Comment (lewisbirks) [reply e2e/tests/localAuthorityMaintainsCaseAfterSubmission_test.js]: See previous

File: e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js (hunk 2 above)
Comment (lewisbirks) [reply e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js]: Would be good, we are just learning how to do the testing at the moment.

File: ccd-definition/AuthorisationCaseEvent.json [hunk 4]
```diff
@@ -291,7 +332,7 @@
     "LiveFrom": "01/01/2017",
     "CaseTypeID": "CARE_SUPERVISION_EPO",
     "CaseEventID": "hearingBookingDetailsGatekeeping",
-    "UserRole": "caseworker-publiclaw-gatekeeper",
```
Comment (toireasam) [inline ccd-definition/AuthorisationCaseEvent.json]: do we not need to give the gatekeeper access to this also?

File: ccd-definition/AuthorisationCaseEvent.json [hunk 5]
```diff
@@ -167,6 +202,13 @@
     "UserRole": "caseworker-publiclaw-courtadmin",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseEventID": "uploadStandardDirectionsGatekeeping",
```
Comment (toireasam) [inline ccd-definition/AuthorisationCaseEvent.json]: do we not need to give access to the gatekeeper role here too?

File: ccd-definition/AuthorisationCaseEvent.json (hunk 4 above)
Comment (lewisbirks) [reply ccd-definition/AuthorisationCaseEvent.json]: Yeah, don't know why that got changed. But it has been decided that it is not in the scope of the project and will be reverted the access for that feature will be given in a new story.

File: ccd-definition/AuthorisationCaseEvent.json (hunk 5 above)
Comment (lewisbirks) [reply ccd-definition/AuthorisationCaseEvent.json]: same as above

File: ccd-definition/CaseEventToFields/uploadDocumentsAfterGatekeeping.json [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseEventID": "uploadDocumentsAfterGatekeeping",
+    "CaseFieldID": "uploadDocuments_paragraph_1",
+    "PageFieldDisplayOrder": 1,
+    "DisplayContext": "READONLY",
+    "PageID": 1,
+    "PageDisplayOrder": 1,
+    "PageColumnNumber": 1,
+    "CallBackURLMidEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/enter-social-work-other/mid-event"
```
Comment (toireasam) [inline ccd-definition/CaseEventToFields/uploadDocumentsAfterGatekeeping.json]: is this callback needed? @akomisarek .. it's in upload documents but not upload documents after submission

File: ccd-definition/CaseEvent.json [hunk 7]
```diff
@@ -457,18 +473,16 @@
...
-    "ID": "createNoticeOfProceedings",
-    "Name": "Create notice of proceedings",
-    "Description": "Create notice of proceedings",
-    "DisplayOrder": 11,
-    "PreConditionState(s)": "Submitted",
-    "PostConditionState": "Submitted",
+    "ID": "uploadDocumentsAfterGatekeeping",
+    "Name": "Documents",
+    "Description": "Upload documents",
+    "DisplayOrder": 14,
```
Comment (akomisarek) [inline ccd-definition/CaseEvent.json]: Numbering is off now, let's fix together.

File: ccd-definition/CaseEventToFields/uploadDocumentsAfterGatekeeping.json (hunk 6 above)
Comment (akomisarek) [reply ccd-definition/CaseEventToFields/uploadDocumentsAfterGatekeeping.json]: I think so, I have added to the submission.

**Comment by toireasam** [comment]
Looks good to me, apart from Adam's comment about the numbering. fix that and should be good to go. good work guys!

File: ccd-definition/AuthorisationCaseState.json [hunk 8]
```diff
@@ -6,6 +6,13 @@
     "UserRole": "caseworker-publiclaw-solicitor",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseStateID": "Open",
+    "UserRole": "caseworker-publiclaw-systemupdate",
+    "CRUD": "CRU"
+  },
```
Comment (akomisarek) [inline ccd-definition/AuthorisationCaseState.json]: Not sure why this was introduced.

File: e2e/tests/gatekeeperAdministersCaseAfterSubmission_test.js [hunk 9]
```diff
@@ -5,7 +5,7 @@ let caseId;
 
 Feature('Gatekeeper Case administration after submission');
```
Comment (akomisarek) [inline e2e/tests/gatekeeperAdministersCaseAfterSubmission_test.js]: Probably that feature and file name is not correct anymore, would be good to have this changed.
